include/fs/fs.h

```diff
@@ -220,6 +220,22 @@ struct fs_statvfs {
 #define FS_FSTAB_DECLARE_ENTRY(node_id)		\
 	extern struct fs_mount_t FS_FSTAB_ENTRY(node_id)
 
+/**
+ * @brief Initialize fs_file_t object
+ *
+ * Initialized the fs_file_t object; the function needs to be invoked
+ * on object before first use with fs_open.
+ *
+ * @param zfp Pointer to file object
+ *
+ */
+inline void fs_file_t_init(struct fs_file_t *zfp)
+{
+	zfp->mp = NULL;
```

Comment:
Would you consider:

    *fp = (struct fs_file_t){
      flags = 0,
    };

This will initialize all fields to zero.  The only reason you need to provide even one is just because it's required syntactically.

This is a bit more flexible in that if you add or rename fields that would be initialized to zero you don't need to touch this function: they'll automatically be zeroed, which wouldn't happen if they're all set individually.

Reply:
@pabigot I have done that before, but this will fail compilation wit
```
tests/subsys/fs/fat_fs_dual_drive/src/main.c:18: undefined reference to `fs_file_t_init
```
and many other places, unless I will define the function as `static inline`.

Reply:
I hadn't noticed: why is this *not* `static inline`?  A non-static inline will fail exactly in that way unless you add this to some implementation file:

```
extern void fs_file_t_init(struct fs_file_t *zfp);
```

This guarantees that there is an extern version available for modules where the inline did not occur.  It comes from [C99](https://en.cppreference.com/w/c/language/inline), and for whatever reason we don't generally use it in Zephyr, preferring to use the old GNU extension `static inline` idiom which forces the inline (or a local callable implementation.

So, you can make this `static inline` so it's like almost every existing header inline function in Zephyr, or add the `extern` declaration to an implementation file.

Reply:
Still: why without static the code with separate assignments linked fine while the one with cast did not?
Doesn't that mean that the assignments were always inlined while the other code could not be inlined in some situations?

Reply:
No idea.  The compiler's free to inline or not based on whatever rules it's using.